src/controls/polyControl.ts

```diff
@@ -52,7 +52,15 @@ export const polyActionHandler = (
     mouseLocalPosition = getLocalPoint(transform, CENTER, CENTER, x, y),
     polygonBaseSize = getSize(poly),
     size = poly._getTransformedDimensions(),
-    sizeFactor = polygonBaseSize.divide(size),
+    // if poly has 0 width or 0 height, NaN's occur so guard for that case
+    sizeFactor = new Point(
+      !size.x || Number.isNaN(size.x)
+        ? 1 / poly.scaleX
+        : polygonBaseSize.x / size.x,
+      !size.y || Number.isNaN(size.y)
+        ? 1 / poly.scaleY
+        : polygonBaseSize.y / size.y
```

Comment:
i have the suspect this won't work with skewing.
for me `sizeFactor = polygonBaseSize.divide(size),` is basically a way to counter transform the skew + scale transformation of the object.
Instead of doing this we could try to calculate the scale/skew/flip matrix, invert it and use transform point on mouseLocalPosition? would that be equivalent?

I m not sure it would be for both the cases of exactBoundingBox and not.

Reply:
I'm not sure how to implement this (it's been a while since I did Matrices and linear algebra). We could separate the MR so that scaling gets fixed if we don't want to try to figure this out right now. The code is basically an expansion of the original code, just providing a value in the 0/NaN case and using the old value otherwise. Are you thinking polygonBaseSize.divide(size) wasn't handling skewing?

Reply:
I think a playwright test can prove you right.
I have an app that worksaround this issue by checking the changed point is not equal to its neighbors and if is it is shifted by `Number.EPSILON` or something

Reply:
Pulling out code to isolate problem. Separate PR can be done for polycontrol issue.